Tests/StarknetTests/Utils/DevnetClient.swift

```diff
@@ -0,0 +1,319 @@
+import Foundation
+
+import Starknet
+
+protocol DevnetClientProtocol {
+    var gatewayUrl: String { get }
+    var feederGatewayUrl: String { get }
+    var rpcUrl: String { get }
+    
+    func start()
+    func close()
+    
+    func prefundAccount(address: Felt)
+    func deployAccount(name: String) -> DeployAccountResult
+    func deployContract(contractPath: String) -> TransactionResult
+    func declareContract(contractPath: String) -> TransactionResult
+    func readAccountDetails(accountName: String) -> AccountDetails
+}
+
+struct AccountDetails: Codable{
+    var privateKey: Felt
+    var publicKey: Felt
+    var address: Felt
+    var salt: Felt
```

Comment:
Let's make these let as well, and also fields in TransactionResult and DeployAccountResult